Add new version syntax based on yml files to the pipeline

JoseEspinosa opened the pull request:
This PR implements the new way of outputing software versions. Software version(s) are now reported for every module imported during a given pipeline execution that is emitted in a `versions.yml` file.
Other minor changes:

*  `params.skip_mirdeep` switch is implemented on `smrnaseq.nf` instead of inside the modules involved in `MIRDEEP2 ` subworkflow.
* Remove hostnames to avoid error in json schema validation in new Nextflow releases (see here nf-core/tools#1304)
* Update mirbase urls
* Update singularity container of `modules/local/bowtie_map_mirna.nf` to match docker one.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] `CHANGELOG.md` is updated.

FriederikeHanssen commented on modules/local/bowtie_map_mirna.nf:
```diff
@@ -1,15 +1,13 @@
+    label 'process_medium'
     publishDir "${params.outdir}",
-        mode: params.publish_dir_mode,
+        mode: params.publish_dir_mode, //TODO check for suffix, is working? needs initOptions?
```
Is this TODO still relevant? If so do you plan on adressing it in a different PR?

JoseEspinosa replied:
I updated the publishDir of all the bowtie local modules here: a5ae9fc

FriederikeHanssen commented on modules/local/bowtie_map_mirna.nf:
```diff
@@ -24,16 +22,14 @@ process BOWTIE_MAP_SEQ {
     script:
-    def software = getSoftwareName(task.process)
-    def process_name = task.process.tokenize(':')[-1]
+    // def process_name = task.process.tokenize(':')[-1] //TODO check if is still needed
```
same here

JoseEspinosa replied:
I updated the publishDir of all the bowtie local modules here: a5ae9fc

FriederikeHanssen commented on modules/local/seqcluster_collapse.nf:
```diff
@@ -1,5 +1,5 @@
 // Import generic module functions
-include { saveFiles; initOptions; getSoftwareName } from './functions'
+include { saveFiles; initOptions; getSoftwareName; getProcessName } from './functions'
```
is initOptions still needed here?

FriederikeHanssen commented on modules/local/trimgalore.nf:
```diff
@@ -1,5 +1,5 @@
 // Import generic module functions
-include { saveFiles; initOptions; getSoftwareName } from './functions'
+include { saveFiles; initOptions; getSoftwareName; getProcessName } from './functions'
```
same here

JoseEspinosa replied:
Addressed here ab5e61a

FriederikeHanssen reviewed (commented):
Nice looks good. Just a couple of things about open TODO statements and having `initOptions` left here and there.

FriederikeHanssen reviewed (approved):
lgtm

JoseEspinosa commented:
I think I have addressed now all @FriederikeHanssen suggestions (very nicely spotted :smile: ) + I updated the files from mirbase in the `nextflow.config` as discussed on slack.
